yeat/cli/config.py

```diff
@@ -8,44 +8,40 @@
 # -------------------------------------------------------------------------------------------------
 
 import json
+from pathlib import Path
 from warnings import warn
 
 
-ALGORITHMS = ("spades", "megahit", "unicycler", "flye", "canu")
-KEYS = ["algorithm", "extra_args"]
+ALGORITHMS = ["spades", "megahit", "unicycler", "flye", "canu"]
+KEYS1 = ["samples", "assemblers"]
+KEYS2 = ["label", "algorithm", "extra_args", "samples"]
 
 
 class AssemblyConfigurationError(ValueError):
     pass
 
 
 class AssemblerConfig:
-    def __init__(self, algorithm, extra_args=""):
-        if algorithm not in ALGORITHMS:
-            raise AssemblyConfigurationError(f"Unsupported assembly algorithm '{algorithm}'")
-        self.algorithm = algorithm
-        self.extra_args = extra_args
-
-    @classmethod
```

Comment:
If you're no longer going to support creation of objects of type `AssemblerConfig`, then you don't need to define a class. These static methods can just be functions defined at the module level.

That said, I think there would be benefit to defining an `AssemblerConfig` class that stores a list of assemblers, a dictionary of samples, and provides functions for validating and accessing the config data.

It's no longer possible to create an object of type `AssemblerConfig`.

Reply:
Took your suggestions and definitely taking the benefits of the class in the new rewrite.